<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneralGameSetupPanel</class>
 <widget class="QWidget" name="GeneralGameSetupPanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>596</width>
    <height>383</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Engine:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="EnginePluginComboBox" name="cboEngine"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelExecutable">
       <property name="text">
        <string>Executable:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="GameExecutablePicker" name="executableInput" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbLockExecutable">
         <property name="toolTip">
          <string>If locked, executable won't change when new config for the same engine is loaded</string>
         </property>
         <property name="text">
          <string>Lock</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Server name:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="leServername">
       <property name="text">
        <string>Started from Doomseeker</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Port:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_81">
       <item>
        <widget class="QSpinBox" name="spinPort">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>10666</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbAllowTheGameToChoosePort">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Allow the game to choose port.</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="5" column="0">
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Game mode:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QComboBox" name="cboGamemode"/>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Map:</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLineEdit" name="leMap"/>
       </item>
       <item>
        <widget class="QLabel" name="lblMapWarning">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>This map isn't present on map list. Game may misbehave.</string>
         </property>
         <property name="pixmap">
          <pixmap resource="../../../../media/resources.qrc">:/icons/exclamation.png</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="labelIwad">
       <property name="text">
        <string>IWAD:</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="IwadPicker" name="iwadPicker" native="true"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Additional WADs and files (check required):</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="WadsPicker" name="wadsPicker" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1,0">
     <item>
      <widget class="QCheckBox" name="cbBroadcastToLAN">
       <property name="text">
        <string>Broadcast to LAN</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="cbBroadcastToMaster">
       <property name="text">
        <string>Broadcast to Master</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="upnpArea" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="cbUpnp">
          <property name="toolTip">
           <string>If checked, the game will try to tell your router to forward necessary ports.</string>
          </property>
          <property name="text">
           <string>UPnP</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinUpnpPort">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;UPnP port. Set this to 0 to let Doomseeker or game decide.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WadsPicker</class>
   <extends>QWidget</extends>
   <header>gui/createserver/wadspicker.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>IwadPicker</class>
   <extends>QWidget</extends>
   <header>gui/createserver/iwadpicker.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>EnginePluginComboBox</class>
   <extends>QComboBox</extends>
   <header>gui/widgets/engineplugincombobox.h</header>
  </customwidget>
  <customwidget>
   <class>GameExecutablePicker</class>
   <extends>QWidget</extends>
   <header>gui/createserver/gameexecutablepicker.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../media/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cboGamemode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>GeneralGameSetupPanel</receiver>
   <slot>onGameModeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>594</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbAllowTheGameToChoosePort</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinPort</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>593</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leMap</sender>
   <signal>textChanged(QString)</signal>
   <receiver>GeneralGameSetupPanel</receiver>
   <slot>updateMapWarningVisibility()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>297</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUpnp</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinUpnpPort</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>513</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setExecutableToDefault()</slot>
  <slot>browseExecutable()</slot>
  <slot>onGameModeChanged(int)</slot>
  <slot>updateMapWarningVisibility()</slot>
 </slots>
</ui>
